pkgs/build-support/dotnet/fetch-nupkg/overrides.nix

```diff
@@ -0,0 +1,49 @@
+{
+  autoPatchelfHook,
+  fontconfig,
+  lib,
+  libICE,
+  libSM,
+  libX11,
+  stdenv,
+  writeText,
+}:
+{
+  # e.g.
+  # "Package.Id" =
+  #   package:
+  #   package.overrideAttrs (old: {
+  #     buildInputs = old.buildInputs or [ ] ++ [ hello ];
+  #   });
+
+  "Avalonia.X11" =
+    package:
+    package.overrideAttrs (
+      old:
+      lib.optionalAttrs stdenv.isLinux {
```

Comment:
Maybe this shouldn't be limited to linux...

Reply:
Because the nuget dependencies aren't platform specific (yet), they need to be pulled in on all platforms.

And those libs actually do exist on darwin.  There's no reason avalonia couldn't connect to an X11 server on darwin.  I don't think it actually would in practice, because it probably has platform specific init code and just does native mac stuff.

I think the most sensible thing to do for now is just `!stdenv.isDarwin`.  We could also filter the input libs by platform.  The only platform I can imagine this being relevant to in the near future is freebsd.